precompile/contracts/rewardmanager/contract.go

```diff
@@ -101,6 +117,20 @@ func allowFeeRecipients(accessibleState contract.AccessibleState, caller common.
 	EnableAllowFeeRecipients(stateDB)
 	packedOutput := []byte{}
 
+	// Add a log to be handled if this action is finalized.
+	if chainCfg.IsDUpgrade(blkCtx.Timestamp()) {
+		topics, data, err := PackFeeRecipientsAllowedEvent(caller)
+		if err != nil {
+			return nil, remainingGas, err
+		}
+		stateDB.AddLog(
+			ContractAddress,
+			topics,
+			data,
+			blkCtx.Number().Uint64(),
+		)
```

Comment:
Hmm since this event has one fixed size argument we should know the number of topics and total size in advance.

imo we should ideally charge before calling pack since packing itself is a variable amount of work based on the arguments that need to be packed into the event. That being said, I'd also agree that it would be good to defer charging that gas to this if branch to make that charge as close to the event packing work as possible.

Reply:
For this PR, since it is handling a fixed size event, I think it makes sense to charge that fixed amount here when it's time to pack the event